emcc.py

```diff
@@ -1443,6 +1446,11 @@ def save_intermediate(name=None, suffix='js'):
         # At minimum remove dead functions etc., this potentially saves a lot in the size of the generated code (and the time to compile it)
         link_opts += shared.Building.get_safe_internalize() + ['-globaldce']
 
+      if cfi:
+        if use_cxx:
+           link_opts.append("-wholeprogramdevirt")
```

Comment:
interesting, what do these flags do? i'm surprised linker options are needed for this functionality, but i'm not that familiar with it so it's probably just my ignorance.)